### ISSUE TITLE
训练奖励持续下降

### ISSUE THREAD

**Issue by xumingw** [yifan123/flow_grpo#226]
非常感谢你的工作和开源！我修改了wan2.2的训练，并且引入了qwen3-235B-vl来对视频质量打分，结果训练发现训练reward持续下跌
以下是我的训练配置，训练机器是H200*8, 请问可能会是什么原因导致的呢

<img width="329" height="252" alt="Image" src="https://github.com/user-attachments/assets/fecf4e69-431f-4299-bdd9-e31944993c08" />

def general_fvd_wan2_2_720p_train():
    config = compressibility()
    config.dataset = os.path.join(os.getcwd(), "dataset/wan_i2v/1_0_video_captions_test_fixpath.json")
    config.test_dataset = os.path.join(os.getcwd(), "dataset/wan_i2v/1_0_video_captions_test_fixpath_test.json")
    config.tb_dir = os.environ.get("OUTPUT_URL", "./tb_dir")

    # config.pretrained.model = "hf_cache/Wan2.1-T2V-14B-Diffusers"
    config.pretrained.model = "./pretrained_models/Wan2.2-TI2V-5B-Diffusers/"
    config.sample.num_steps = 20  # 20
    config.sample.eval_num_steps = 50  # 50
    config.sample.guidance_scale=5.0
    config.run_name = "wan2.2_flow_grpo_train_qwen"
    
    config.height = 704
    config.width = 1280
    config.frames = 21 # 
    config.sample.train_batch_size = 1
    config.sample.num_image_per_prompt = 4 # 4 # 12
    config.sample.num_batches_per_epoch = 4
    config.sample.sample_time_per_prompt = 4 # 4
    config.sample.test_batch_size = 1

    config.train.batch_size = config.sample.train_batch_size
    config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch * config.sample.sample_time_per_prompt // 2 if (config.sample.num_batches_per_epoch * config.sample.sample_time_per_prompt) > 1 else 1
    config.train.num_inner_epochs = 1
    config.train.timestep_fraction = 0.99
    # kl loss
    config.train.beta = 0.004
    config.train.learning_rate = 1e-4
    config.train.clip_range=1e-3
    config.train.cfg = True
    # kl reward
    # KL reward and KL loss are two ways to incorporate KL divergence. KL reward adds KL to the reward, while KL loss, introduced by GRPO, directly adds KL loss to the policy loss. We support both methods, but KL loss is recommended as the preferred option.
    config.sample.kl_reward = 0.0
    # We also support using SFT data in RL training for supervised learning to prevent quality drop, but this option was unused
    config.train.sft=0.0
    config.train.sft_batch_size=3
    # Whether to use the std of all samples or the current group's.
    config.sample.global_std = False
    config.train.ema = True
    config.mixed_precision = "bf16"
    config.diffusion_loss = True
    # A large num_epochs is intentionally set here. Training will be manually stopped once sufficient
    config.num_epochs = 100000
    config.save_freq = 50 # epoch
    config.eval_freq = 50000000
    config.save_dir = os.path.join(os.environ.get("OUTPUT_URL", "./log"), f'logs/video_fvd/{config.run_name}')
    config.obs_upload_path = os.environ.get("OBS_UPLOAD_PATH", None)
    config.resume_from = None
    config.reward_fn = {
        "qwen3vl_vq_remote": 1.0,
    }
    
    config.prompt_fn = "fvd"

    config.per_prompt_stat_tracking = True
    return config

**Comment by Jayce-Ping** [comment]
感觉有两种可能的原因：
1. Wan2.2这个模型不太好训练上去，我在实验过程中发现相同的配置对wan2.1 work但对wan2.2不work。感觉你可以先试试wan2.1能不能正常涨分。
2. reward打分不可靠。能具体分享一下你是怎么打分的吗

**Comment by xumingw** [comment]
> 感觉有两种可能的原因：
> 
> 1. Wan2.2这个模型不太好训练上去，我在实验过程中发现相同的配置对wan2.1 work但对wan2.2不work。感觉你可以先试试wan2.1能不能正常涨分。
> 2. reward打分不可靠。能具体分享一下你是怎么打分的吗

感谢回复！我目前正在尝试wan2.1,  我试了ocr能正常涨分，正在试新的reward, 目前是把视频提交给qwen3-vl，然后写了一段比较简单的prompt要求他根据视频视觉质量按0-5打分

**Comment by RobbinW** [comment]
@xumingw hi ，我也在尝试使用其他reward 在wan 2.1 和2.2 系列上训练~ 请问你有用qwen3-vl 把reward训上去吗

**Comment by xumingw** [comment]
> [@xumingw](https://github.com/xumingw) hi ，我也在尝试使用其他reward 在wan 2.1 和2.2 系列上训练~ 请问你有用qwen3-vl 把reward训上去吗

没有，现在只是没有再下降，但是也没有提升

**Comment by Jayce-Ping** [comment]
ocr能涨分的话，我猜测可能是mllm打分不是很solid. 试试把temperature设为0呢，或者先试试[UnifiedReward Think](https://github.com/CodeGoat24/UnifiedReward)？

**Comment by Darius-H** [comment]
> 感觉有两种可能的原因：
> 
> 1. Wan2.2这个模型不太好训练上去，我在实验过程中发现相同的配置对wan2.1 work但对wan2.2不work。感觉你可以先试试wan2.1能不能正常涨分。
> 2. reward打分不可靠。能具体分享一下你是怎么打分的吗

你尝试了wan2.1 ocr可以work吗，我测试readme里面的wan2.1训练，但是得到的reward曲线是这样的，看上去也不work，是训练步数还不够吗：
<img width="989" height="490" alt="Image" src="https://github.com/user-attachments/assets/7cd74a40-487e-4a8d-8d74-a31974103fa7" />

**Comment by xumingw** [comment]
> > 感觉有两种可能的原因：
> > 
> > 1. Wan2.2这个模型不太好训练上去，我在实验过程中发现相同的配置对wan2.1 work但对wan2.2不work。感觉你可以先试试wan2.1能不能正常涨分。
> > 2. reward打分不可靠。能具体分享一下你是怎么打分的吗
> 
> 你尝试了wan2.1 ocr可以work吗，我测试readme里面的wan2.1训练，但是得到的reward曲线是这样的，看上去也不work，是训练步数还不够吗： <img alt="Image" width="989" height="490" src="https://private-user-images.githubusercontent.com/49788663/555739204-7cd74a40-487e-4a8d-8d74-a31974103fa7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzIxNjMxMzcsIm5iZiI6MTc3MjE2MjgzNywicGF0aCI6Ii80OTc4ODY2My81NTU3MzkyMDQtN2NkNzRhNDAtNDg3ZS00YThkLThkNzQtYTMxOTc0MTAzZmE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjI3VDAzMjcxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2MzJkZTdlNTMwOWI1MWYwMWMzYTc4YjI2MGI3YTFjOTFmNzY1YmMxZDYxOGJjZWY2Nzc0NGI1NzhmZjQ5ZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VoHKNtx47UWnGfSEaZ35GiHRIqqxRhVNsLZ82bFR28U">

wan2.1 ocr我测试是正常的

**Comment by y0sh1no** [comment]
> 非常感谢你的工作和开源！我修改了wan2.2的训练，并且引入了qwen3-235B-vl来对视频质量打分，结果训练发现训练reward持续下跌 以下是我的训练配置，训练机器是H200*8, 请问可能会是什么原因导致的呢
> 
> <img alt="Image" width="329" height="252" src="https://private-user-images.githubusercontent.com/89380537/544677970-fecf4e69-431f-4299-bdd9-e31944993c08.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzIzNzkxNjksIm5iZiI6MTc3MjM3ODg2OSwicGF0aCI6Ii84OTM4MDUzNy81NDQ2Nzc5NzAtZmVjZjRlNjktNDMxZi00Mjk5LWJkZDktZTMxOTQ0OTkzYzA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzAxVDE1Mjc0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1ODkwZTI3ZjllMWNhOGYxZmY0NGUyYjBkYjg2NjM4YmQwZDgwYmFlMzRiZDY2MzRkMmFkMGZiMjRjNGU0ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OfBUhGN0O0xuB89YYdBkZWciPndyHxBZBFR4aJq3vo4">
> def general_fvd_wan2_2_720p_train(): config = compressibility() config.dataset = os.path.join(os.getcwd(), "dataset/wan_i2v/1_0_video_captions_test_fixpath.json") config.test_dataset = os.path.join(os.getcwd(), "dataset/wan_i2v/1_0_video_captions_test_fixpath_test.json") config.tb_dir = os.environ.get("OUTPUT_URL", "./tb_dir")
> 
> ```
> # config.pretrained.model = "hf_cache/Wan2.1-T2V-14B-Diffusers"
> config.pretrained.model = "./pretrained_models/Wan2.2-TI2V-5B-Diffusers/"
> config.sample.num_steps = 20  # 20
> config.sample.eval_num_steps = 50  # 50
> config.sample.guidance_scale=5.0
> config.run_name = "wan2.2_flow_grpo_train_qwen"
> 
> config.height = 704
> config.width = 1280
> config.frames = 21 # 
> config.sample.train_batch_size = 1
> config.sample.num_image_per_prompt = 4 # 4 # 12
> config.sample.num_batches_per_epoch = 4
> config.sample.sample_time_per_prompt = 4 # 4
> config.sample.test_batch_size = 1
> 
> config.train.batch_size = config.sample.train_batch_size
> config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch * config.sample.sample_time_per_prompt // 2 if (config.sample.num_batches_per_epoch * config.sample.sample_time_per_prompt) > 1 else 1
> config.train.num_inner_epochs = 1
> config.train.timestep_fraction = 0.99
> # kl loss
> config.train.beta = 0.004
> config.train.learning_rate = 1e-4
> config.train.clip_range=1e-3
> config.train.cfg = True
> # kl reward
> # KL reward and KL loss are two ways to incorporate KL divergence. KL reward adds KL to the reward, while KL loss, introduced by GRPO, directly adds KL loss to the policy loss. We support both methods, but KL loss is recommended as the preferred option.
> config.sample.kl_reward = 0.0
> # We also support using SFT data in RL training for supervised learning to prevent quality drop, but this option was unused
> config.train.sft=0.0
> config.train.sft_batch_size=3
> # Whether to use the std of all samples or the current group's.
> config.sample.global_std = False
> config.train.ema = True
> config.mixed_precision = "bf16"
> config.diffusion_loss = True
> # A large num_epochs is intentionally set here. Training will be manually stopped once sufficient
> config.num_epochs = 100000
> config.save_freq = 50 # epoch
> config.eval_freq = 50000000
> config.save_dir = os.path.join(os.environ.get("OUTPUT_URL", "./log"), f'logs/video_fvd/{config.run_name}')
> config.obs_upload_path = os.environ.get("OBS_UPLOAD_PATH", None)
> config.resume_from = None
> config.reward_fn = {
>     "qwen3vl_vq_remote": 1.0,
> }
> 
> config.prompt_fn = "fvd"
> 
> config.per_prompt_stat_tracking = True
> return config
> ```

您好，请问如何修改flowgrpo以支持wan22? 我刚刚接触视频生成的grpo相关工作，您可以提供一些建议/指导嘛？

**Comment by Jayce-Ping** [comment]
@y0sh1no [Flow-Factory](https://github.com/X-GenGroup/Flow-Factory/tree/main)有一版实现可以供你参考。
